### PR TITLE
Shared Script

### PR DESCRIPTION
**Pull request by freznva**

### CODE REVIEW

File: code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp [hunk 1]
```diff
@@ -64,6 +64,8 @@ ResourceScriptingComponent::ResourceScriptingComponent(Resource* resource)
 #endif
 			);
 
+			auto sharedScripts = metaData->GetEntries("shared_script")
```
Comment (blattersturm) [inline code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp]: Does this even compile? There's a _missing semicolon_.

File: code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp [hunk 2]
```diff
@@ -78,6 +80,15 @@ ResourceScriptingComponent::ResourceScriptingComponent(Resource* resource)
 					}
 				}
 
+				for (auto& sharedScript : sharedScripts)
```
Comment (blattersturm) [inline code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp]: Code duplication is annoying. It's probably better to make an outer loop that iterates over lists, such as:

```cpp
for (auto& list : { blahScripts, sharedScripts })
{
    for (auto& script : list)
    {
        // ...
    }
}
```

File: code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp [hunk 3]
```diff
@@ -205,6 +218,19 @@ void ResourceScriptingComponent::CreateEnvironments()
 					}
 				}
 			}
+
+			for (auto& sharedScript : sharedScripts)
```
Comment (blattersturm) [inline code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp]: Same code duplication issue applies here.

File: code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp (hunk 1 above)
Comment (freznva) [reply code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp]: Yes, it compiles. But idk how it came out with a semicolon

**Comment by freznva** [comment]
wow, what have i done...

**Pull request closed without merging**